Zend/zend_language_scanner.l

```diff
@@ -1250,6 +1258,69 @@ static void copy_heredoc_label_stack(void *void_heredoc_label)
 	zend_ptr_stack_push(&SCNG(heredoc_label_stack), (void *) new_heredoc_label);
 }
 
+/* Check that { }, [ ], ( ) are nested correctly */
+static void report_bad_nesting(char opening, int opening_lineno, char closing)
+{
+	char buf[256];
+	int  used = 0;
+
+	if (opening == '$') {
+		used = snprintf(buf, sizeof(buf), "Unclosed '${'");
+	} else {
+		used = snprintf(buf, sizeof(buf), "Unclosed '%c'", opening);
+	}
+
+	if (opening_lineno != CG(zend_lineno)) {
+		used += snprintf(buf + used, sizeof(buf) - used, " on line %d", opening_lineno);
+	}
+
+	if (closing) { 	/* 'closing' will be 0 if at end of file */
+		used += snprintf(buf + used, sizeof(buf) - used, " does not match '%c'", closing);
```

Comment:
`used +=` is not needed, but I still do it in case someone comes and adds more code below. Don't want any smashed stacks.